modules/core/src/controllers/orthographic-controller.js

```diff
@@ -0,0 +1,165 @@
+import Controller from './controller';
+import ViewState from './view-state';
+
+import {Vector2, clamp} from 'math.gl';
+
+// TODO: not sure whether we should have these constants
+// as part of viewport props
+const MOVEMENT_SPEED = 10; // per keyboard click
+const MIN_ZOOM = 0.1;
+const MAX_ZOOM = 10;
+
+class OrthographicState extends ViewState {
+  constructor({
+    /* Viewport arguments */
+    width, // Width of viewport
+    height, // Height of viewport
+    offset, // Offset to the origin
+    zoom, // Zoom level of the view
+
+    /** Interaction states */
+    startPanPosition,
+    startPanOffset,
+    startRotatePosition,
+    startRotateOffset
+  }) {
+    super({
+      width,
+      height,
+      offset,
+      zoom
+    });
+    this._interactiveState = {
+      startPanPosition,
+      startPanOffset,
+      startRotatePosition,
+      startRotateOffset
+    };
+  }
+
+  /* Public API */
+
+  getInteractiveState() {
+    return this._interactiveState;
+  }
+
+  /**
+   * Start panning
+   * @param {[Number, Number]} pos - position on screen where the pointer grabs
+   */
+  panStart({pos}) {
+    const {offset} = this._viewportProps;
+    return this._getUpdatedState({
+      startPanPosition: pos,
+      startPanOffset: offset
+    });
+  }
+
+  /**
+   * Pan
+   * @param {[Number, Number]} pos - position on screen where the pointer is
+   */
+  pan({pos}) {
+    const {startPanPosition, startPanOffset} = this._interactiveState;
+    const delta = new Vector2(pos).subtract(startPanPosition);
+    return this._getUpdatedState({
+      offset: new Vector2(startPanOffset).subtract(delta)
+    });
+  }
+
+  /**
+   * End panning
+   * Must call if `panStart()` was called
+   */
+  panEnd() {
+    return this._getUpdatedState({
+      startPanPosition: null,
+      startPanOffset: null
+    });
+  }
+
+  /**
+   * Start rotating
+   * @param {[Number, Number]} pos - position on screen where the pointer grabs
+   */
+  rotateStart({pos}) {
+    return this;
+  }
+
+  /**
+   * Rotate
+   * @param {[Number, Number]} pos - position on screen where the pointer is
+   */
+  rotate({deltaScaleX, deltaScaleY}) {
+    return this;
+  }
+
+  /**
+   * End rotating
+   * Must call if `rotateStart()` was called
+   */
+  rotateEnd() {
+    return this;
+  }
+
+  /**
+   * Zoom
+   * @param {Number} scale - a number between [0, 1] specifying the accumulated
+   *   relative scale.
+   */
+  zoom({scale}) {
+    const {zoom} = this._viewportProps;
+    const newZoom = clamp(zoom * scale, MIN_ZOOM, MAX_ZOOM);
+    return this._getUpdatedState({
+      zoom: newZoom
```

Comment:
This is not zooming around the cursor position. Can we match the behavior of other controllers?

Reply:
I notice it, too. Will make a change soon